plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/rename/RenameElementRefactoringProcessor.java

```diff
@@ -90,19 +90,22 @@ private void createChildChanges(final CompositeChange change) {
 				? new DataTypeInstanceSearch(dtEntry).performSearch()
 				: new BlockTypeInstanceSearch(typeEntry).performSearch();
 		final var eChild = getChildByURI(typeEntry.getType(), elementURI);
-		String oldName = ""; //$NON-NLS-1$
-		if (eChild instanceof final IInterfaceElement varDecl) {
-			oldName = varDecl.getName();
-		}
 
 		for (final EObject eObject : result) {
-			if (eObject instanceof FBNetworkElement) {
-				change.add(new ReconnectPinChange(EcoreUtil.getURI(eObject), FBNetworkElement.class, newName, oldName));
+			if (eObject instanceof final FBNetworkElement element
+					&& eChild instanceof final IInterfaceElement varDecl) {
+				createRenameVarDeclChanges(change, element, varDecl);
 			}
 		}
 
 	}
 
+	private void createRenameVarDeclChanges(final CompositeChange change, final FBNetworkElement element,
+			final IInterfaceElement varDecl) {
```

Comment:
I believe this is also used for other interface elements and not just `VarDeclaration`s? In that case, I would suggest to change the naming to reflect that.